src/Cli/dotnet/commands/dotnet-run/Program.cs

```diff
@@ -24,6 +24,35 @@ public static RunCommand FromParseResult(ParseResult parseResult)
                 parseResult = ModifyParseResultForShorthandProjectOption(parseResult);
             }
 
+            // if the application arguments contain any binlog args then we need to remove them from the application arguments and apply
+            // them to the restore args.
+            // this is because we can't model the binlog command structure in MSbuild in the System.CommandLine parser, but we need
+            // bl information to synchronize the restore and build logger configurations
+            var applicationArguments = parseResult.GetValue(RunCommandParser.ApplicationArguments).ToList();
+
+            var binlogArgs = new List<string>();
+            var nonBinLogArgs = new List<string>();
+            foreach (var arg in applicationArguments)
+            {
+
+                if (arg.StartsWith("/bl:") || arg.Equals("/bl")
```

Comment:
if you just did `arg.Cotnains("/bl"` and ignored the `:` variant` would that still work and simplify the check a bit?

Reply:
it would potentially capture user-application arguments that started with `bl` but were otherwise distinct from MSBuild's `/bl` arg.